src/code/z_sram_NES.c

```diff
@@ -28,58 +28,77 @@ typedef struct PersistentCycleFlags {
 /**
  * Array of bitwise flags which won't be turned off on a cycle reset (will persist between cycles)
  */
-PersistentCycleFlags sPersistentCycleFlags[SCENE_MAX] = {
+PersistentCycleSceneFlags sPersistentCycleSceneFlags[SCENE_MAX] = {
 #include "tables/scene_table.h"
 };
 
 #undef DEFINE_SCENE
 #undef DEFINE_SCENE_UNSET
 
-// TODO: figure out a way to use the WEEKEVENTREG defines here
+#define WEEK_FLG2(flg) (2 << (2 * BIT_FLAG_TO_SHIFT(flg)))
+#define WEEK_FLG(flg) (3 << (2 * BIT_FLAG_TO_SHIFT(flg)))
```

Comment:
I feel like we could do better on the names for these ("upper" and "both", for example), but it'd be very difficult to get it any shorter. It's probably not worth it anyway, one per line seems fine.

I will say that the way they are now threw me off when I was reading the array, I assumed "2" checked for both bits.

Reply:
The logical names for these would be something like `PERSISTENT_WEEKEVENTREG(...)` which I was worried about length, but I think it's fine actually.

Reply:
Could also change the `2` to `ALT` since this is dealing with bits, so the "2" might be misinterpreted